plugins/module_utils/network/sonic/config/lldp_interfaces/lldp_interfaces.py

```diff
@@ -0,0 +1,374 @@
+        """
+        commands = []
+        requests = []
+        if not want:
+            commands = have
+            requests.extend(self.get_delete_lldp_interfaces_complete_requests(commands))
+        else:
+            commands = get_diff(want, diff)
+            for command in commands:
+                requests.extend(self.get_delete_specific_lldp_interfaces_param_requests(command, have))
+
+        if len(requests) == 0:
+            commands = []
+
+        if commands:
+            commands = update_states(commands, "deleted")
+
+        return commands, requests
+
+    def _state_replaced(self, diff, want, have):
+        """ The command generator when state is replaced
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        if not diff:
+            return commands, requests
+        commands = want
+        for command in commands:
+            # Get interface name and make all configs to default in that interface
+            command = {"name": command["name"]}
+            requests.extend(self.get_delete_specific_lldp_interfaces_param_requests(command, have))
+
+        for command in commands:
+            requests.extend(self.get_modify_specific_lldp_interfaces_param_requests(command))
+
+        if len(requests) > 0:
+            commands = update_states(commands, "replaced")
+        return commands, requests
+
+    def _state_overridden(self, want, have, diff):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        if not diff:
+            return commands, requests
+        commands = have
+        # Default all existing interface configuration
+        requests.extend(self.get_delete_lldp_interfaces_complete_requests(commands))
+
+        commands_overridden = want
+        for command in commands_overridden:
+            requests.extend(self.get_modify_specific_lldp_interfaces_param_requests(command))
+        if len(requests) > 0:
+            commands = update_states(commands_overridden, "overridden")
+
+        return commands, requests
+
+    def get_delete_lldp_interfaces_complete_requests(self, have):
+        """Get requests to delete all existing LLDP global
+        configurations in the chassis
+        """
+        default_dict = {'tlv_select': {'power_management': True}, 'med_tlv_select': {'network_policy': True, 'power_management': True}, 'enable': True}
+        requests = []
+        conf = copy.deepcopy(have)
+        for cfg in conf:
+            del cfg['name']
+            if default_dict != cfg:
+                return [{'path': self.lldp_interfaces_path, 'method': DELETE}]
+        return requests
```

Comment:
The login shows either one "cfg" in deleting  request or request is []. Please check.

Reply:
I see that the logic is intended to delete all LLDP interface configuration with one REST request if at least one interface is found with non-default configuration. This looks okay to me.

There is, however, currently no test case for this. Please add one.

Reply:
Added a new case for this scenario

Reply:
The posted test result looks correct.

Resolved.